src/CFG-based-system.typ

```diff
@@ -1 +1,178 @@
-= CFG Based System
\ No newline at end of file
+#import "./proof-tree.typ": *
+#import "./vars.typ": *
+= CFG Based System
+
+Basic rules for Uniqueness and Borrowing won't change here,
+but adapted to fit the CFG representation of Kotlin.
+
+The key gap between the CFG version and the type rules is at the function call.
+Type checking at function calls need adjustment so that it can be done in control flow order.
+The support for recursive function all is alse missing is previous typing rules.
+
+== Formalization
+
+We follow the original lattice for control flow analysis.
+#figure(image(width: 25%, "./img/lattice.svg"), caption: [Lattice in Control Flow Analysis])<CFA-lattice>
+
+The context for CFA $Gamma$ is defined as: a mapping from path to the lattice:
+$ Gamma : p -> U $, $U$ for uniqueness and borrowing level.
+
+In Kotlin CFG, $"$"i = "eval" e$, is a common pattern that appears everywhere, we distinguish this case with assignment statement when rhs is a path: define context $Delta$ for case $"$"i = "eval" p$ temp variables to the corresponding path and lattice, to make it simple here, we define it as a list:
+$ Delta : (i, p, U) $
+When the rhs is not a variable or path, we store $(i, *, U)$.
```

Comment:
This is very confusing to read.  Please split it up: what is U?  What is Γ?  What is Δ?  What kinds of nodes does the CFG have?  Don't define Δ by explaining what happens to it in one particular case, treat all these things step-by-step.